zipline/pipeline/factors/technical.py

```diff
@@ -144,6 +148,205 @@ def compute(self, today, assets, out, close, volume):
+        # )
+        regression_factor = SingleRegressionFactor(
+            asset=Equity(24),
+            returns_length=10,
+            regression_length=5,
+        )
+        alpha = regression_factor.alpha
+        beta = regression_factor.beta
+
+    The result of computing ``alpha`` from 2017-03-17 to 2017-03-22 gives::
+
+                      AAPL    MSFT     FB
+        2017-03-17       0    .011   .003
+        2017-03-20       0   -.004   .004
+        2017-03-21       0    .007   .006
+        2017-03-22       0    .002   .008
+
+    And the result of computing ``beta`` from 2017-03-17 to 2017-03-22 gives::
+
+                      AAPL    MSFT     FB
+        2017-03-17       1      .3   -1.1
+        2017-03-20       1      .2     -1
+        2017-03-21       1     -.3     -1
+        2017-03-22       1     -.3    -.9
+
+    Note that AAPL's column for alpha is all 0's and for beta is all 1's, as
+    the regression line of a data set with itself is simply the function y = x.
+
+    To understand how each of the other values were calculated, take for
+    example MSFT's ``alpha`` and ``beta`` values on 2017-03-17 (.011 and .3,
+    respectively). These values are the result of running a linear regression
+    predicting MSFT's returns from AAPL's returns, using values starting at
+    2017-03-17 and looking back 5 days. That is, the regression was run with
+    x = [-.03, -.02, -.01, 0, .01] and y = [.03, -.03, .02, -.02, .04], and it
+    produced a slope of .3 and an intercept of .011.
+
+    See Also
+    --------
+    :class:`zipline.pipeline.factors.technical.CorrelationFactor`
+    """
+    params = ('asset', 'dependent')
+
+    def __new__(cls,
+                asset,
+                returns_length,
+                regression_length,
+                dependent=False,
+                mask=NotSpecified,
+                dtype=NotSpecified,
+                missing_value=NotSpecified,
+                **kwargs):
+        new_instance = super(SingleRegressionFactor, cls).__new__(
+            cls,
+            asset=asset,
+            dependent=dependent,
+            inputs=[Returns(window_length=returns_length)],
+            outputs=['alpha', 'beta'],
+            window_length=regression_length,
+            mask=mask,
+            dtype=dtype,
+            missing_value=missing_value,
+            **kwargs
+        )
+        return new_instance
+
+    def compute(self, today, assets, out, returns, asset, dependent):
+        asset_col = where(assets == asset.sid)[0][0]
+        my_asset = returns[:, asset_col]
+        for i in range(len(out)):
+            other_asset = returns[:, i]
+            if dependent:
+                regr_results = linregress(y=my_asset, x=other_asset)
+            else:
+                regr_results = linregress(y=other_asset, x=my_asset)
+            # `linregress` returns its results in the following order:
+            # slope, intercept, r-value, p-value, stderr
```

Comment:
I don't know how much memory/computational complexity this would add. It would be nice to have the other 3 columns, but I feel like a small percentage of use cases would need them. Generally you'd run your statistical tests at a higher level in the research environment, or a few broad statistical tests on your overall portfolio motion. Getting a p-value for each asset is gonna result in a ton of multiple comparisons bias, but could be useful for dropping assets on which the regression clearly doesn't work.

 If it doesn't slow down the method much, of it there's a way to flag whether or not the factor returns these, I would say do it.